Updated source files to avoid namespace duplication

laf opened the pull request:
## Pre-Request Checklist
<!-- Not all items apply to each PR, but a great PR addresses all applicable items. -->

- [ ] Passes rubocop code analysis (try `rubocop --auto-correct`)
- [ ] Tests added or adapted (try `rake test`)
- [ ] Changes are reflected in the documentation
- [ ] User-visible changes appended to [CHANGELOG.md](/CHANGELOG.md)

## Description
This possibly fixes issue https://github.com/ytti/oxidized/issues/2154. I've got a limited test environment I've run this in and without changing the config file this works ok for me.

manager.rb may need more doing to it but this worked for me.

laf commented:
@robertcheramy Would you be able to review this change?

robertcheramy commented:
Yes, I will look at your PR but must work on other issues and older PR first.

michieltimmers commented:
FYI. I tested the pull request and it fixed the problem I had. I can now have both Librenms has http source, and also have input with http.

@laf Thanks!!!

robertcheramy commented on lib/oxidized/manager.rb:
```diff
@@ -8,7 +8,7 @@ class << self
         require File.join dir, file + '.rb'
         klass = nil
-        [Oxidized, Object].each do |mod|
+        [Oxidized, Source, Object].each do |mod|
```
It helped me understanding the code better when using the full module path (`Oxidized::Source`) even if `Source` is  correct. What do you think?

Could you add a comment describing what this is doing, something like:
```
# Search the object in the different namespaces
# - Most objects are in the namespace Oxidized
# - Source objects have their own namespace (Oxidized::Source)
# - Models have no namespace (Object)
```

laf replied:
I'm happy to take direction on using the full module path, I've updated the PR to use it and added your comment in.

robertcheramy reviewed (changes requested):
1. Could you add a unit test for your change in the manager class under /spec/manager_spec.rb. I suggest:
```
require_relative 'spec_helper'

describe Oxidized::Manager do
  before(:each) do
    Oxidized.asetus = Asetus.new
    Oxidized.asetus.cfg.debug = false
    Oxidized.setup_logger
  end

  describe '#add_source' do
    it 'loads a source when available' do
      Oxidized.config.source.csv.file = '/some/path'
      result = Oxidized.mgr.add_source('csv')
      _(result['csv']).must_equal Oxidized::Source::CSV
    end

    it 'returns nil when the source is not available' do
      result = Oxidized.mgr.add_source('XXX')
      _(result).must_be_nil
    end
  end
end
```
I would have included it into your PR, bud you didn't give the oxidized maintainer the right to modify your PR.

2. Please document your change in CHANGELOG.md

laf commented:
> 1. Could you add a unit test for your change in the manager class under /spec/manager_spec.rb. I suggest:
> 
> ```
> require_relative 'spec_helper'
> 
> describe Oxidized::Manager do
>   before(:each) do
>     Oxidized.asetus = Asetus.new
>     Oxidized.asetus.cfg.debug = false
>     Oxidized.setup_logger
>   end
> 
>   describe '#add_source' do
>     it 'loads a source when available' do
>       Oxidized.config.source.csv.file = '/some/path'
>       result = Oxidized.mgr.add_source('csv')
>       _(result['csv']).must_equal Oxidized::Source::CSV
>     end
> 
>     it 'returns nil when the source is not available' do
>       result = Oxidized.mgr.add_source('XXX')
>       _(result).must_be_nil
>     end
>   end
> end
> ```
> 
> I would have included it into your PR, bud you didn't give the oxidized maintainer the right to modify your PR.
> 
> 2. Please document your change in CHANGELOG.md

Sorry for the delay on this one @robertcheramy. I've updated the CHANGELOG and also added the test file in with a small modification. I appreciate the feedback.

robertcheramy commented:
Thanks for the corrections. I've found two new problems for the PR:
- It relies on the order in which add_source and add_input are called.
- It does not resolve the conflict with the Output http

I'm currently fixing these problems and will then merge into master. You do not need to modify this PR.

robertcheramy commented:
Manualy rebased/squashed/merged into master: 73c8861204c7536276336ddda5eda62398217634 and b3f3f9542d5196e044871a2d24753c1c23b2eb1a
Closes: #2154

laf commented:
> Thanks for the corrections. I've found two new problems for the PR:
> 
> * It relies on the order in which add_source and add_input are called.
> * It does not resolve the conflict with the Output http
> 
> I'm currently fixing these problems and will then merge into master. You do not need to modify this PR.

Thanks @robertcheramy. I appreciate the feedback and help on this. Glad it got sorted in the end :)
